data/data/bootstrap/baremetal/files/usr/local/bin/coredns.sh

```diff
@@ -25,15 +25,18 @@ if ! podman inspect "$COREDNS_IMAGE" &>/dev/null; then
     podman pull "$COREDNS_IMAGE"
 fi
 MATCHES="$(sudo podman ps -a --format "{{.Names}}" | awk '/coredns$/ {print $0}')"
-if [[ -z "$MATCHES" ]]; then
-    /usr/bin/podman create \
-        --name coredns \
-        --volume /etc/coredns:/etc/coredns:z \
-        --network host \
-        --env CLUSTER_DOMAIN="$CLUSTER_DOMAIN" \
-        --env CLUSTER_NAME="$CLUSTER_NAME" \
-        --env NUM_DNS_MEMBERS="$NUM_DNS_MEMBERS" \
-        --env API_VIP="$API_VIP" \
-        "${COREDNS_IMAGE}" \
-            --conf /etc/coredns/Corefile
+if [[ ! -z "$MATCHES" ]]; then
+    # Remove old pod, or you can get storage for container removed
+    # errors on restart if the container exits unexpectedly
+    podman rm -f coredns
 fi
+/usr/bin/podman create \
```

Comment:
why is `--rm                                       Remove container (and pod if created) after exit` not used here ?

Reply:
Maybe that would be a better way, thanks for pointing it out.

Indeed, the basis of this bug I'm trying to fix is that the exited container can't be safely restarted, so we need to remove it before any systemd restart happens.

I'll add --rm (although I'd prefer to leave the additional script cleanup logic as well, so we can be absolutely sure on start that no stale containers exist, running or otherwise, so the end result is the same).

Reply:
Now done